### PR TITLE
Add an assert_asserts macro, test assertions in currently tested functions

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

**Review by kgaillot — COMMENTED** [review]
This is a big step forward, we really have a nice test framework now :)

Update Pacemaker Development and it's good to go

File: include/crm/common/unittest_internal.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/* A cmocka-like assert macro for use in unit testing.  This one verifies that
+ * an expression aborts through CRM_ASSERT, erroring out if that is not the case.
+ *
+ * This macro works by running the expression in a forked child process with core
+ * dumps disabled (CRM_ASSERT calls abort(), which will write out a core dump).
+ * The parent waits for the child to exit and checks why.  If the child received
+ * a SIGABRT, the test passes.  For all other cases, the test fails.
+ */
+#define assert_asserts(expr) \
```
[the remark this answers is not in the document]
Comment (clumens) [reply include/crm/common/unittest_internal.h:35]: Updated.

**Comment by clumens** [comment]
This one is now good to go.

File: doc/sphinx/Pacemaker_Development/helpers.rst [hunk 2]
```diff
@@ -238,6 +224,17 @@ is no ``lib/common/tests/acls`` directory.
...
      check_PROGRAMS = pcmk_acl_required_test
 
+* Double check that ``-I$(top_srcdir)/lib/common`` is present in ``AM_CPPFLAGS``.
+
+* Double check the settings of variables at the top of ``Makefile.am``.  The
+  following block should be present:
+
+  .. code-block:: none
+
+     LDADD = $(top_builddir)/lib/common/libcrmcommon_test.la
```
Comment (nrwahl2) [inline doc/sphinx/Pacemaker_Development/helpers.rst:234]: `-lcmocka`?

Comment (clumens) [reply doc/sphinx/Pacemaker_Development/helpers.rst:234]: Yes, added a patch to #2793 to fix this.

**Pull request merged**